src/dispatch/plugins/dispatch_slack/middleware.py

```diff
@@ -112,6 +112,15 @@ def message_context_middleware(request: BoltRequest, context: BoltContext, next:
     next()
 
 
+def reaction_context_middleware(context: BoltContext, payload: dict, next: Callable) -> None:
```

Comment:
@kevgliss @wssheldon I added a middleware for reactions. There's probably an opportunity to combine this one with the `command_context_middleware` and `message_context_middleware`. The code is almost the same, just different exception error message.

Reply:
Now that I see this, we may want to rethink the global error handler logic, specifically how we pick the message for the user:

TLDR; message is determined by exception type, not message raised with the exception. 

```python
def build_and_log_error(...snip) -> str:
    if isinstance(error, RoleError):
        message = build_role_error_message(payload)
        logger.info(error)
  
    ...snip
    
    elif isinstance(error, ContextError):
        message = build_context_error_message(payload, error)
        logger.info(error)
        
    ...snip

    else:
        guid = str(uuid.uuid4())
        message = build_unexpected_error_message(guid)
        logger.exception(error, extra=dict(slack_interaction_guid=guid))

    return message
```

Reply:
Should we do it before pushing to prod?